storage/s3.go

```diff
@@ -43,6 +46,9 @@ const (
 
 	// Google Cloud Storage endpoint
 	gcsEndpoint = "storage.googleapis.com"
+
+	// the key of the object metadata which is used to handle retry decision on NoSuchUpload error
+	metadataKeyRetryCode = "S5cmd-Retry-Code"
```

Comment:
```suggestion
	metadataKeyRetryCode = "S5cmd-Retry-Code"
```

How about `s5cmd-upload-retry-id`?